src/prefect/cli/cloud/__init__.py

```diff
@@ -594,23 +594,44 @@ async def set(
                 "Unable to authenticate. Please ensure your credentials are correct."
             )
 
-    if workspace_handle:
-        # Search for the given workspace
-        for workspace in workspaces:
-            if workspace.handle == workspace_handle:
-                break
+        if workspace_handle:
+            # Search for the given workspace
+            for workspace in workspaces:
+                if workspace.handle == workspace_handle:
+                    break
+            else:
+                exit_with_error(f"Workspace {workspace_handle!r} not found.")
         else:
-            exit_with_error(f"Workspace {workspace_handle!r} not found.")
-    else:
-        workspace = prompt_select_from_list(
-            app.console,
-            "Which workspace would you like to use?",
-            [(workspace, workspace.handle) for workspace in workspaces],
-        )
+            # don't force two choices if there's not that many workspaces
+            if len(workspaces) > 10:
+                accounts = await client.read_accounts()
```

Comment:
You might be able to get away with not making the extra 2 API calls. `workspaces = await client.read_workspaces()` should have all the information needed with a little bit of bucketing